virttest/qemu_devices/qcontainer.py

```diff
@@ -1082,8 +1087,14 @@ def pflash_handler(firmware_name, cmd):
                 else:
                     img_path, img_name = os.path.split(img_info.get("filename"))
 
-                    pflash_vars_name = "%s_%s_VARS.fd" % (self.vmname, img_name)
-                    pflash_vars_path = os.path.join(img_path, pflash_vars_name)
+                    pflash_vars_name = "%s_%s_%s_VARS.fd" % \
+                        (self.vmname, '_'.join(img_name.split('.')),
```

Comment:
For raw block devices, the image_name is more like `sdX` or others, so they don't have a format.

Reply:
```
a = 'sdX'
'_'.join(a.split('.'))
Out[3]: 'sdX'
```
If there is no format, just display 'sdX' which is acceptable for me.
Or do you think is it better to redefine it?  
in `if not os.access(pflash_vars_path, os.W_OK):`

```
pflash_vars_name = "%s_%s_%s_%s_VARS.fd" % \
                        (self.vmname, img_name, params['image_backend'],
                         img_info.get("format"))
```